spikesorters/yass/yass.py

```diff
@@ -0,0 +1,280 @@
+                        cortical data. 
+                        For more information see https://www.biorxiv.org/content/10.1101/2020.03.18.997924v1
+                        """
+
+    installation_mesg = """\nTo Install and Use Yass follow the wiki: 
+                            https://github.com/paninski-lab/yass/wiki
+                        """
+
+    def __init__(self, **kargs):
+        print (**kargs)
+        #BaseSorter.__init__(self, rec, output_folder)
+        self.params = self._default_params
+        
+    @classmethod
+    def is_installed(cls):
+        #return HAVE_YASS
+        #return check_if_installed(cls.kilosort2_path)
+        return self.HAVE_YASS
+    
+    @staticmethod
+    def get_sorter_version():
+        return yass.__version__
+
+    # NEED TO CHANGE THIS FOR YASS ALSO
+    # n_chan = recording.get_num_channels()
+    # n_frames = recording.get_num_frames()
+    # chunk_size = 2 ** 24 // n_chan
+    
+    # also make a default config file for
+    # https://github.com/SpikeInterface/spikesorters/blob/master/spikesorters/spyking_circus/config_default.params
+    # {} reserved for params that need to be updated at run time;
+    
+    # this function parses params and creates config file and binary data and geometry file also
+    # and saving everything to the output folder
+    # Cat: this function changes the minimum required default values; 
+    def _setup_recording(self, recording, output_folder):
+        p = self.params
+        source_dir = Path(output_folder).parent
+
+        #################################################################
+        #################### UPDATE ROOT FOLDER #########################
+        #################################################################
+        # float(self._recording.sample_rate.rescale('Hz').magnitude)
+        self.params['data']['root_folder'] = output_folder
+        #self.params['data']['geometry'] = 'geom.csv'
+        
+        #################################################################
+        #################### GEOMETRY FILE GENERATION ###################
+        #################################################################
+        # save prb file
+        # note: only one group here, the split is done in basesorter
+        probe_file_csv = os.path.join(output_folder,'geom.csv')
+        probe_file_txt = os.path.join(output_folder,'geom.txt')
+        # ALESSIO .saveto probe saved .prb file; have to d thisourselves.
+        #  
+        adjacency_radius = -1
+        recording.save_to_probe_file(probe_file_csv, 
+                                     grouping_property=None,
+                                     radius=adjacency_radius)
+        
+        import csv
+
+        with open(probe_file_csv) as csv_file:
+            csv_reader = csv.reader(csv_file, delimiter=',')
+            geom_txt = np.float32(np.vstack(csv_reader))
+            np.savetxt(probe_file_txt, geom_txt)
```

Comment:
I don't anderstand why you save first to csv and then to txt.
You can use recording.get_channel_location to store directly to txt.

Reply:
Ok, fixing that now.